### PR TITLE
Add limits to global config 

### PR DESCRIPTION
**Pull request by vesari**
This PR adds the following limits to the global config:

- body_size_limit
- sample_limit
- target_limit
- label_limit
- label_name_length_limit
- label_value_length_limit

When scrape configs do not set these parameters, they get the default from the global config instead. Conversely, if a scrape config sets one of these limits, it will override the global one.

Fixes https://github.com/prometheus/prometheus/issues/11494

I've been given permission by @gouthamve to complete work on this issue.

### CODE REVIEW

**Comment by roidelapluie** [comment]
This should be done on Validate() and the Validate function should probably take the full global config as parameter now.

**Comment by vesari** [comment]
> This should be done on Validate() and the Validate function should probably take the full global config as parameter now.

@roidelapluie please check my latest revision. Thanks!

**Review by SuperQ — APPROVED** [review]
Nice

**Comment by xiu** [comment]
Hi! Is there anything blocking this one from getting merged?

File: config/config.go [hunk 1]
```diff
@@ -389,6 +389,24 @@ type GlobalConfig struct {
 	QueryLogFile string `yaml:"query_log_file,omitempty"`
 	// The labels to add to any timeseries that this Prometheus instance scrapes.
 	ExternalLabels labels.Labels `yaml:"external_labels,omitempty"`
+	// An uncompressed response body larger than this many bytes will cause the
+	// scrape to fail. 0 means no limit.
```
Comment (bwplotka) [inline config/config.go:393]: I guess we need "0 means no limit." everywhere on those below?

**Review by bwplotka — APPROVED** [review]
Nice, I think it's fine. Should we also update documentation in this PR, or separate? (it's in the same repo in /docs)

Thanks!

**Comment by vesari** [comment]
> Nice, I think it's fine. Should we also update documentation in this PR, or separate? (it's in the same repo in /docs)
> 
> Thanks!
@bwplotka thank you, I'll be happy to update the documentation in this PR

File: config/config.go (hunk 1 above)
Comment (vesari) [reply config/config.go:393]: I've done it for every new limit field in the global config, and I've made sure to also make the scrape config limits consistent in this regard.

**Comment by vesari** [comment]
@bwplotka I've added the documentation for the new global fields, PTAL.

**Review by bwplotka — APPROVED** [review]
Thanks, LGTM!

cc @roidelapluie

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**